app/src/main/java/io/homeassistant/companion/android/widgets/ButtonWidget.kt

```diff
@@ -0,0 +1,210 @@
+
+import android.app.PendingIntent
+import android.appwidget.AppWidgetManager
+import android.appwidget.AppWidgetProvider
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.os.Handler
+import android.util.Log
+import android.widget.RemoteViews
+import io.homeassistant.companion.android.R
+import io.homeassistant.companion.android.common.dagger.GraphComponentAccessor
+import io.homeassistant.companion.android.domain.integration.IntegrationUseCase
+import io.homeassistant.companion.android.domain.widgets.WidgetUseCase
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+class ButtonWidget : AppWidgetProvider() {
+    companion object {
+        private const val TAG = "ButtonWidget"
+        private const val CALL_SERVICE =
+            "io.homeassistant.companion.android.widgets.ButtonWidget.CALL_SERVICE"
+        internal const val RECEIVE_DATA =
+            "io.homeassistant.companion.android.widgets.ButtonWidget.RECEIVE_DATA"
+
+        internal const val EXTRA_DOMAIN = "EXTRA_DOMAIN"
+        internal const val EXTRA_SERVICE = "EXTRA_SERVICE"
+        internal const val EXTRA_SERVICE_DATA = "EXTRA_SERVICE_DATA"
+        internal const val EXTRA_LABEL = "EXTRA_LABEL"
+    }
+
+    @Inject
+    lateinit var integrationUseCase: IntegrationUseCase
+    @Inject
+    lateinit var widgetStorage: WidgetUseCase
+
+    private val mainScope: CoroutineScope = CoroutineScope(Dispatchers.Main + Job())
+
+    override fun onUpdate(
+        context: Context,
+        appWidgetManager: AppWidgetManager,
+        appWidgetIds: IntArray
+    ) {
+        // There may be multiple widgets active, so update all of them
+        for (appWidgetId in appWidgetIds) {
+            mainScope.launch {
+                try {
+                    val intent = Intent(context, ButtonWidget::class.java).apply {
+                        action = CALL_SERVICE
+                        putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId)
+                    }
+
+                    val views = RemoteViews(context.packageName, R.layout.widget_button).apply {
+                        setOnClickPendingIntent(
+                            R.id.widgetImageButton,
+                            PendingIntent.getBroadcast(
+                                context,
+                                appWidgetId,
+                                intent,
+                                PendingIntent.FLAG_UPDATE_CURRENT
+                            )
+                        )
+                        setTextViewText(
+                            R.id.widgetLabel,
+                            widgetStorage.loadLabel(appWidgetId)
+                        )
+                    }
+                    appWidgetManager.updateAppWidget(appWidgetId, views)
+                } catch (e: Exception) {
```

Comment:
What is this try catch trying to catch?

Reply:
Nothing, apparently.  I think I copied it from my other `mainScope.launch {...}` block and didn't think closely enough about why the try catch was there.  Removed in 36cb729.